python/tank/templatekey.py

```diff
@@ -47,6 +47,12 @@ def __init__(self,
         self.length = length
         self._last_error = ""
 
+        # check that the key name doesn't contain invalid characters
```

Comment:
Yepp, ok, I can totally see it being down both ways. Mostly, I am trying to be consistent with how things are being done in the rest of toolkit. Maybe we can discuss and see what the group thinks. Personally, I guess I am thinking pragmatically about it - the core is a small chunk of code - fundamentally encapsulated by the separation between engines, apps, core etc - and within the core itself, I find it handy that there is one place you can go where you find important constants that concern the overall behaviour. In any case, it's more a matter of being *consisntent* with the currently established patterns here. If we want to change those patterns, I gues that's a bigger discussion.